src/vcpkg-test/configparser.cpp

```diff
@@ -361,27 +453,49 @@ TEST_CASE ("BinaryConfigParser azblob provider", "[binaryconfigparser]")
     }
     {
         auto parsed = create_binary_providers_from_configs_pure("x-azblob,https://azure/container,sas,read", {});
-        REQUIRE(parsed.has_value());
+        auto state = parsed.value_or_exit(VCPKG_LINE_INFO);
+
+        REQUIRE(state.binary_cache_providers == std::set<std::string>{"azblob"});
+        REQUIRE(url_temp.get_url(state) == std::string{"https://azure/container/{sha}.zip?sas"});
+        REQUIRE(state.secrets == std::vector<std::string>{"sas"});
+        REQUIRE(!state.archives_to_read.empty());
     }
     {
         auto parsed = create_binary_providers_from_configs_pure("x-azblob,https://azure/container,sas,write", {});
-        REQUIRE(parsed.has_value());
+        auto state = parsed.value_or_exit(VCPKG_LINE_INFO);
+
+        REQUIRE(state.binary_cache_providers == std::set<std::string>{"azblob"});
+        REQUIRE(url_temp.get_url(state) == std::string{"https://azure/container/{sha}.zip?sas"});
```

Comment:
This one should be asserting that the read set is empty and the write set is only that one